clippy_lints/src/loops/explicit_into_iter_loop.rs

```diff
@@ -1,24 +1,27 @@
 use super::EXPLICIT_INTO_ITER_LOOP;
 use clippy_utils::diagnostics::span_lint_and_sugg;
 use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::{match_trait_method, paths};
 use rustc_errors::Applicability;
 use rustc_hir::Expr;
 use rustc_lint::LateContext;
 use rustc_middle::ty::TyS;
 
-pub(super) fn check(cx: &LateContext<'_>, args: &'hir [Expr<'hir>], arg: &Expr<'_>) {
-    let receiver_ty = cx.typeck_results().expr_ty(&args[0]);
-    let receiver_ty_adjusted = cx.typeck_results().expr_ty_adjusted(&args[0]);
-    if !TyS::same_type(receiver_ty, receiver_ty_adjusted) {
+pub(super) fn check(cx: &LateContext<'_>, self_arg: &'hir Expr<'hir>, call_expr: &Expr<'_>) {
+    if !(TyS::same_type(
+        cx.typeck_results().expr_ty(self_arg),
+        cx.typeck_results().expr_ty_adjusted(self_arg),
+    ) && match_trait_method(cx, call_expr, &paths::INTO_ITERATOR))
```

Comment:
This looks like a weird change to me style wise. I'd prefer to keep those two variables and just add the `match_trait_method` check. This makes the `if` condition way harder to read.